include/shared/config/conf.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef CONF_H
+#define CONF_H
+
+#include <map>
+#include <string>
+#include <iostream>
+
+#define NEG -1 // Negative value when it's false
+
+enum Configuration {
+   SERVER_IP,
+   SERVER_PORT,
+   CLIENT_IP,
+   CLIENT_PORT,
+   LAST
+};
+
+class Config {
+   static Config* m_instance;
+   std::string const m_base_conf = "../config/config.conf";
+   std::string m_user_conf;
+   std::string m_user_dir;
+
+   bool           compare_files();
+
+public:
+   Config();
+   
+   static Config& get_instance();
+   
+   enum ValidationLog { 
+      Info, 
+      Error, 
+      Debug, 
+      Critical 
+   };
+
+   std::string    to_string(Configuration t_value) const; // enum to string
```

Comment:
Let's group the public methods together to make the class more readable and have a better sense of what exactly is the public API of this class.